in-vehicle-digital-twin/src/digitaltwin_impl.rs

```diff
@@ -110,62 +98,27 @@ impl DigitalTwin for DigitalTwinImpl {
 }
 
 impl DigitalTwinImpl {
-    /// This function assumes that an array of resources has been provided and that each resource in the array needs to be registered.
-    ///
-    /// # Arguments
-    /// * `dtdl` - The DTDL for the array.
-    #[allow(unused_variables)]
-    fn register_each_one(&self, dtdl: &str) -> Result<(), String> {
-        let doc: Value = match serde_json::from_str(dtdl) {
-            Ok(json) => json,
-            Err(error) => return Err(format!("Failed to parse the DTDL due to: {error:?}")),
-        };
-
-        match doc {
-            Value::Array(array) => {
-                for v in array.iter() {
-                    self.register_one(v)?
-                }
-            }
-            _ => return Err(String::from("An unexpected item was encountered in the DTDL.")),
-        };
-
-        Ok(())
-    }
-
-    /// Register the resource specified in the the JSON doc.
+    /// Register the entity.
     ///
     /// # Arguments
-    /// * `doc` - The JSON doc that specifies the entity.
-    fn register_one(&self, doc: &Value) -> Result<(), String> {
-        let dtdl = match serde_json::to_string_pretty(&doc) {
-            Ok(content) => content,
-            Err(error) => {
-                return Err(format!("Failed to make the DTDL pretty due to: : {error:?}"))
-            }
-        };
-
-        let mut parser = ModelParser::new();
-        let json_texts = vec![dtdl];
-
-        let model_dict_result = parser.parse(&json_texts);
-        if let Err(error) = model_dict_result {
-            return Err(format!("Failed to parse the DTDL due to: {error:?}"));
-        }
-        let model_dict = model_dict_result.unwrap();
-
+    /// * `entity` - The entity.
+    fn register_entity(&self, entity_access_info: EntityAccessInfo) -> Result<(), String> {
         // This block controls the lifetime of the lock.
         {
-            let mut lock: RwLockWriteGuard<HashMap<String, Value>> = self.entity_map.write();
-            for id in model_dict.keys() {
-                lock.insert(id.to_string(), doc.clone());
-            }
+            let mut lock: RwLockWriteGuard<HashMap<String, EntityAccessInfo>> =
```

Comment:
again I would suggest removing the type annotation here if possible

Reply:
This is aa stylistic preference.  I prefer to leave it in for cases like this where it helps me to understand what I am dealing with.